### PR TITLE
Triage: update issue forms and automate label updates

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

GitHub recently updated their issue predefs, and now allow you to create issue forms. Those are interactive, rather than just relying on inline comments in the issue body placeholder, so there is more chance that they'll actually be used.

In this PR, I've done the following:
﻿- Add new GitHub issue form handlers.
- Remove old templates that will be replaced by our new setup.
- Listen to changes to issues in our Gardening action (issues opened or reopened).
- Gardening: add new task to automatically add labels to new issues, based on the standardized content of the new issues.

This should help us better triage new issues in the monorepo, and thus help make sure that plugin authors, regardless of the plugin, can get to their issues faster, without having to rely on manual triage.

#### Jetpack product discussion

* No

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Check for typos
* You can check the new forms by viewing the files directly:
    * https://github.com/Automattic/jetpack/blob/add/issue-forms-and-triage/.github/ISSUE_TEMPLATE/bug-report.yml
    * https://github.com/Automattic/jetpack/blob/add/issue-forms-and-triage/.github/ISSUE_TEMPLATE/enhancement.yml
    * https://github.com/Automattic/jetpack/blob/add/issue-forms-and-triage/.github/ISSUE_TEMPLATE/feature-request.yml
* I'm not sure we can check the new task before this gets merged.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+name: Bug Report
+description: Report issues with one of our products
```
Comment (kraftbj) [inline .github/ISSUE_TEMPLATE/bug-report.yml:2]:
```suggestion
description: Report issues with one of our products.
```

**Review by kraftbj — COMMENTED** [review]
small change, I'll commit it and approve it.

**Review by kraftbj — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/enhancement.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      label: What
+      description: Add a concise description of the change being suggested.
+      placeholder: eg. I would like a new dropdown at <xyz>...
+    validations:
+      required: true
+  - type: textarea
+    id: how
+    attributes:
+      label: How
+      description: If applicable, add screenshots, mockup, aminations and/or videos to help illustrate what the change could look like.
```
Comment (sdixon194) [inline .github/ISSUE_TEMPLATE/enhancement.yml:37]:
```suggestion
      description: If applicable, add screenshots, mockup, animations, and/or videos to help illustrate what the change could look like.
```

File: .github/ISSUE_TEMPLATE/feature-request.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      label: What
+      description: Add a concise description of the feature being suggested.
+      placeholder: eg. I would like a new dropdown at <xyz>...
+    validations:
+      required: true
+  - type: textarea
+    id: how
+    attributes:
+      label: How
+      description: If applicable, add screenshots, mockup, aminations and/or videos to help illustrate how the feature could be done.
```
Comment (sdixon194) [inline .github/ISSUE_TEMPLATE/feature-request.yml:37]:
```suggestion
      description: If applicable, add screenshots, mockup, animations, and/or videos to help illustrate how the feature could be done.
```

**Review by sdixon194 — CHANGES_REQUESTED** [review]
Some typo changes.

**Review by sdixon194 — APPROVED** [review]

**Review by sdixon194 — APPROVED** [review]

**Comment by jeherve** [comment]
Here is the end result:

image

It looks like GitHub adds its own security entry since we have a [Security policy file](https://github.com/Automattic/jetpack/blob/master/SECURITY.md). Do you think I should remove the extra Security link I added at the bottom?

And a test issue: #20735 

The Gardening action appears to be working:

https://github.com/Automattic/jetpack/actions/runs/1142282250

image

**Comment by kraftbj** [comment]
I could be convinced either way, but I vote for removing it. Keeps it cleaner and one less thing to change if we ever move from H1.

**Comment by jeherve** [comment]
👍 Done in #20738.

**Pull request merged**